pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -941,11 +937,11 @@ func (vs *VSphere) HasClusterID() bool {
 // Notification handler when node is registered.
 func (vs *VSphere) NodeRegistered(node *v1.Node) {
 	glog.V(4).Infof("Node Registered: %+v", node)
-	vs.nodeManager.registerNode(node)
+	vs.nodeManager.RegisterNode(node)
 }
 
 // Notification handler when node is unregistered.
 func (vs *VSphere) NodeUnregistered(node *v1.Node) {
 	glog.V(4).Infof("Node Unregistered: %+v", node)
-	vs.nodeManager.unregisterNode(node)
+	vs.nodeManager.UnRegisterNode(node)
```

Comment:
What happens if UnRegisterNode() is unable to unregister node?

Reply:
This is just an in memory operation, so the likely hood of this happening is non-existent unless if there is a bug in the map implementation.

Reply:
Sorry. My query was for registerNode()

Reply:
It returns an error. Later, while attach/detach ops it will be anyways rediscover the node

Reply:
UnRegisterNode will not throw at all.